src/MuxManager.cpp

```diff
@@ -470,4 +472,79 @@ void MuxManager::generateServerMac(uint16_t serverId, std::array<uint8_t, ETHER_
     }
 }
 
+// ---> processWarmRestartFlag
+//
+// process warm restart flag
+//
+void MuxManager::processWarmRestartFlag(const std::string &flag)
+{
+    mWarmRestartFlag = flag;
+    
+    if(flag == "true") {
+        MUXLOGINFO("Detected a warm restart is on, starting reconciliation timer.");
+        startWarmRestartReconciliationTimer();
+    }
+}
+
+// ---> updateWarmRestartReconciliationCount(int increment);
+//
+// update warm restart reconciliation count
+//
+void MuxManager::updateWarmRestartReconciliationCount(int increment)
+{
+    MUXLOGDEBUG(increment);
+
+    boost::asio::io_service &ioService = mStrand.context();
+
+    ioService.post(mStrand.wrap(boost::bind(
+        &MuxManager::handleUpdateReconciliationCount,
+        this,
+        increment
+    )));
+}
+
+// ---> handleUpdateReconciliationCount(int increment);
+//
+// handler of updating reconciliation port count
+// 
+void MuxManager::handleUpdateReconciliationCount(int increment)
+{
+    MUXLOGDEBUG(mPortReconciliationCount);
+
+    mPortReconciliationCount += increment;
+
+    if(mPortReconciliationCount == 0) {
+        mReconciliationTimer.cancel();
+    } 
+}
+
+// ---> startWarmRestartReconciliationTimer
+//
+// start warm restart reconciliation timer
+// 
+void MuxManager::startWarmRestartReconciliationTimer()
+{
+    mReconciliationTimer.expires_from_now(boost::posix_time::seconds(
```

Comment:
Is this a blocking call? If yes, then won't all ports just wait for timeout rather than checking for reconciliation while the timer is running?

Reply:
This is a non-blocking asynchronous wait.